### PR TITLE
Updated eslint.

### LINKED ISSUE #25: Update eslint version
Project eslint version is outdated. Apart from lint rules, the current version also doesn't have prettier set up which may lead to inconsistent formatting.

### PR DESCRIPTION
**Pull request by wadamek65**
I've split the changes into a few commits to make the review easier. The first one is purely formatting changes. The second one is some typing adjustments due to new error rules in the eslint configuration. The type changes are also a small step towards #24 .

#### Note
Had to ignore the `__mongo__` and `__tests__` directories as they contained way too many errors. I've created a separate issue for this: #26

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,21 +12,25 @@
...
-    "eslint": "5.4.0",
-    "eslint-config-prettier": "3.0.1",
-    "eslint-config-vazco": "4.0.0",
-    "eslint-plugin-babel": "5.1.0",
-    "eslint-plugin-prettier": "2.6.2",
-    "eslint-plugin-react": "7.11.1",
-    "eslint-plugin-vazco": "1.0.0",
+    "eslint": "7.17.0",
+    "eslint-config-prettier": "7.1.0",
+    "eslint-config-vazco": "6.1.0",
```
Comment (radekmie) [inline package.json:16]: How about upgrading `package-json.lock` to v2 and require npm@7 in the `engines` section? It'd let us get rid of all of these lines here.

File: packages/sparrowql/src/build.ts [hunk 2]
```diff
@@ -30,301 +50,390 @@ export type Options = {
...
   limit?: number;
-  match?: any;
-  projection?: any;
+  match?: unknown;
+  projection?: Record<string, unknown>;
   relation?: Relation;
   skip?: number;
-  sort?: any;
-}
+  sort?: Record<string, 0 | 1>;
```
Comment (radekmie) [inline packages/sparrowql/src/build.ts]:
```suggestion
  sort?: Record<string, -1 | 1>;
```

File: packages/sparrowql/src/build.ts [hunk 3]
```diff
@@ -30,301 +50,390 @@ export type Options = {
...
+    } else {
+      steps.splice(
+        indexMatch + 2,
+        0,
+        steps.splice(
+          steps.findIndex(step => 'limit' in step),
+          1,
+        )[0],
+      );
+    }
```
Comment (radekmie) [inline packages/sparrowql/src/build.ts:187]: We can leave it for now, but at least add a `TODO` here - this code is now barely readable.

**Review by radekmie — CHANGES_REQUESTED** [review]

File: packages/sparrowql/src/build.ts (hunk 3 above)
Comment (wadamek65) [reply packages/sparrowql/src/build.ts:187]: As this code is repeated three times here what do you think about exporting it to a function like for example
```ts
const spliceSteps = (indexOffset: number, stepName: 'skip' | 'limit') => {
      const [element] = steps.splice(
          steps.findIndex(step => stepName in step),
          1,
        )
      steps.splice(indexMatch + indexOffset, 0, element);
}
```

File: package.json (hunk 1 above)
Comment (wadamek65) [reply package.json:16]: Sounds good to me, changed.

File: packages/sparrowql/src/build.ts (hunk 2 above)
Comment (wadamek65) [reply packages/sparrowql/src/build.ts]: Fixed.

File: packages/sparrowql/src/build.ts (hunk 3 above)
Comment (radekmie) [reply packages/sparrowql/src/build.ts:187]: Yeah, something like that. My proposition:
```ts
const wrapAtPhase = (index: number, stepName: 'limit' | 'skip') => {
  const pivot = steps.findIndex(step => stepName in step);
  steps.splice(index, 0, steps.splice(pivot, 1)[0]);
}
```

Comment (wadamek65) [reply packages/sparrowql/src/build.ts:187]: Looks good! Changed.

**Review by radekmie — APPROVED** [review]

**Pull request merged**